apophis/cmds/roll.py

```diff
@@ -0,0 +1,208 @@
+""" Roll Command """
+from cmds.command import Command
+from dataclasses import dataclass
+from typing import Any
+from typing import List
+from typing import Optional
+from typing import Union
+import random
+import re
+
+
+@dataclass
+class Expression:
+    value: Union['Dice', 'Constant', 'Add']
+
+
+@dataclass
+class Dice:
+    rolls: int
+    faces: int
+
+
+@dataclass
+class Constant:
+    c: int
+
+
+@dataclass
+class Add:
+    left: Expression
+    right: Expression
+
+
+def dice(rolls: int, faces: int) -> Expression:
+    return Expression(Dice(rolls, faces))
+
+
+def constant(c: int) -> Expression:
+    return Expression(Constant(c))
+
+
+def add(lhs: Expression, rhs: Expression) -> Expression:
+    return Expression(Add(lhs, rhs))
+
+
+def evaluate(expr: Expression) -> List[int]:
+    value = expr.value
+    if isinstance(value, Constant):
+        return [value.c]
+    elif isinstance(value, Dice):
+        xs = [0] * value.rolls
```

Comment:
You can use `range` instead: https://docs.python.org/3/library/functions.html#func-range